### DISCUSSION TITLE
I can't pull my own repository, turns out I cannot ping github.com either

### DISCUSSION THREAD

**Discussion by santitomatis** [community/community#27498] [Repositories] [answered]
So when I try to pull my own repository I get 
"ssh: Could not resolve hostname main: Name or service not known
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists."

The repository does exists and I'm pretty sure I do have acces rigths, since I configured my ssh keys following github's documentation and I even did it twice (deleting previous attempt)

The ssh key agent is working and the key is in my keychain, so just being curious I tried pinging github.com and a I get this:
"PING github.com (20.201.28.151) 56(84) bytes of data.
^C
--- github.com ping statistics ---
4 packets transmitted, 0 received, 100% packet loss, time 3146ms"
Note that I can ping otther servers like google.com

Using the nslookup github.com command:
"Server:         172.31.144.1
Address:        172.31.144.1#53

Non-authoritative answer:
Name:   github.com
Address: 20.201.28.151 "

Note that I'm running Ubuntu 20.04 in WSL2.
i'd appreciate any help

**Comment by airtower-luna** [comment, marked as answer]
> "ssh: Could not resolve hostname main: Name or service not known
> fatal: Could not read from remote repository.

This looks like some error in your repository or SSH configuration. If you're trying to pull over SSH the hostname of the server should not be `main`. Correct would be `github.com`, or possibly `ssh.github.com` (if you [use SSH over the HTTPS port](https://docs.github.com/en/authentication/troubleshooting-ssh/using-ssh-over-the-https-port)).

Not being able to ping sounds like *something* interfering along the way (e.g. a misconfigured firewall). The IP address you see for `github.com` seems correct, it is part of the IP ranges https://api.github.com/meta lists in the `web` section.

I'd start with debugging the SSH connection. Try this command to get debug info:

```sh
ssh -vT git@github.com
```

**Reply by santitomatis** [reply]
Thanks a lot for helping! I think that you're right and maybe there was an error in my local directory .git configuration or something, but I just created another directory (local of course), cloned my repository there and now it works just fine, so kinda solved it

**Reply by airtower-luna** [reply]
I'm glad to hear it works, even if the problem remains a mystery! :grin:
